src/BottomNavigation/BottomNavigationItem.js

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import shallowEqual from 'recompose/shallowEqual';
+import EnhancedButton from '../internal/EnhancedButton';
+
+function getStyles(props, context) {
+  const {selected} = props;
+
+  const {muiTheme} = context;
+  const {bottomNavigation} = muiTheme;
+
+  const color = selected ? bottomNavigation.selectedColor :
+                           bottomNavigation.unselectedColor;
+  const styles = {
+    root: {
+      transition: 'all 0.3s',
+      paddingTop: selected ? '6px' : '8px',
+      paddingBottom: '10px',
+      paddingLeft: '12px',
+      paddingRight: '12px',
+      minWidth: '80px',
+      maxWidth: '168px',
+    },
+    label: {
+      fontSize: selected ? bottomNavigation.selectedFontSize :
+                           bottomNavigation.unselectedFontSize,
+      transition: 'all 0.3s',
+      color: color,
+    },
+    icon: {
+      display: 'block',
+    },
+    iconColor: color,
+  };
+
+  return styles;
+}
+
+class BottomNavigationItem extends React.Component {
+  static muiName = 'BottomNavigationItem';
+
+  static propTypes = {
+    /**
+     * Set the icon representing the view for this item.
+     */
+    icon: React.PropTypes.node,
+
+    /**
+     * Set the label describing the view for this item.
+     */
+    label: React.PropTypes.string,
+
+    /**
+     * Override the inline-styles of the root element.
+     */
+    style: React.PropTypes.object,
+  };
+
+  static contextTypes = {
+    muiTheme: React.PropTypes.object.isRequired,
+  };
+
+  shouldComponentUpdate(nextProps, nextState) {
```

Comment:
As this composent is talking nodes as input.
Most of the time, we are going to waste CPU cycles in this method.
Could we remove it? 
It would be most efficient to have the pure logic upper in the tree, most likely in user land.